shopinvader_customer_price/models/shopinvader_backend.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2020 Camptocamp SA (http://www.camptocamp.com).
+# @author Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, tools
+
+
+class ShopinvaderBackend(models.Model):
+
+    _inherit = "shopinvader.backend"
+
+    cart_pricelist_partner_field_id = fields.Many2one(
+        comodel_name="ir.model.fields",
+        domain=[
+            ("model", "=", "res.partner"),
+            ("ttype", "=", "many2one"),
+            ("relation", "=", "product.pricelist"),
+        ],
+        help="Set the partner pricelist that will be used for the cart. "
+        "WARNING: by changing this you might have a mismatch "
+        "between the prices showed on the cart "
+        "and the ones showed on product details. "
+        "The default pricelist will still be used for products' indexes.",
+    )
+
+    @tools.ormcache("partner.id", "self.cart_pricelist_partner_field_id.id")
+    def _get_cart_pricelist_id(self, partner):
+        if self.cart_pricelist_partner_field_id:
+            pricelist = partner[self.cart_pricelist_partner_field_id.name]
+            return pricelist.id
+        return None
+
+    def _get_cart_pricelist(self, partner=None):
+        pricelist = super()._get_cart_pricelist(partner)
+        if partner:
+            pricelist = self._get_partner_pricelist(partner) or pricelist
+        return pricelist
+
+    def _get_partner_pricelist(self, partner):
```

Comment:
discussed w/ Seb: move this method to core and let it return the default pricelist.
hen hook here and in s_sale_profile See #807